x-pack/plugins/security/public/views/management/roles_grid/components/roles_grid_page.tsx

```diff
@@ -0,0 +1,271 @@
+              callback={this.handleDelete}
+            />
+          ) : null}
+
+          {
+            // @ts-ignore missing rowProps typedef
+            <EuiInMemoryTable
+              itemId="name"
+              columns={this.getColumnConfig(intl)}
+              selection={{
+                itemId: 'name',
+                selectable: (role: Role) => !role.metadata || !role.metadata._reserved,
+                selectableMessage: (selectable: boolean) =>
+                  !selectable ? 'Role is reserved' : undefined,
+                onSelectionChange: (selection: Role[]) => this.setState({ selection }),
+              }}
+              pagination={{
+                initialPageSize: 20,
+                pageSizeOptions: [10, 20, 30, 50, 100],
+              }}
+              items={this.getVisibleRoles()}
+              loading={roles.length === 0}
+              search={{
+                toolsLeft: this.renderToolsLeft(),
+                box: {
+                  incremental: true,
+                },
+                onChange: (query: Record<string, any>) => {
+                  this.setState({
+                    filter: query.queryText,
+                  });
+                },
+              }}
+              sorting={{
+                sort: {
+                  field: 'name',
+                  direction: 'asc',
+                },
+              }}
+              // @ts-ignore missing rowProps typedef
+              rowProps={() => {
+                return {
+                  'data-test-subj': 'roleRow',
+                };
+              }}
+              isSelectable
+            />
+          }
+        </EuiPageContentBody>
+      </EuiPageContent>
+    );
+  };
+
+  private getColumnConfig = (intl: InjectedIntl) => [
+    {
+      field: 'name',
+      name: intl.formatMessage({
+        id: 'xpack.security.management.roles.nameColumnName',
+        defaultMessage: 'Role',
+      }),
+      sortable: true,
+      truncateText: true,
+      render: (name: string, record: Role) => {
+        return (
+          <span>
+            <EuiLink data-test-subj="roleRowName" href={getRoleManagementHref(name)}>
+              {name}
+            </EuiLink>
+            {!isRoleEnabled(record) && (
+              <EuiIconTip
+                type="alert"
+                size="xs"
+                color="warning"
+                data-test-subj="disabledRoleTip"
+                title={
+                  <FormattedMessage
+                    id="xpack.security.management.roles.disabledTooltip"
+                    defaultMessage="This role is currently disabled. You may only view or delete it."
+                  />
+                }
+              />
+            )}
+          </span>
+        );
+      },
+    },
+    {
+      field: 'metadata',
+      name: intl.formatMessage({
+        id: 'xpack.security.management.roles.reservedColumnName',
+        defaultMessage: 'Reserved',
+      }),
+      sortable: false,
```

Comment:
It would be nice to allow the reserved column to be sortable so they can sort to have all their non-reserved options first.

<img src="https://d.pr/free/i/3R2ixy+" width="50%" />